rust/passwords/src/sha512crypt.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2025 Arista Networks, Inc.
+// Use of this source code is governed by the Apache License 2.0
+// that can be found in the LICENSE file.
+
+use sha_crypt::{CryptError, Sha512Params, sha512_crypt_b64};
+
+pub fn sha512_crypt(password: String, salt: String) -> Result<String, CryptError> {
+    // Setting rounds to 5000 which is the default for sha512crypt
+    let params = Sha512Params::new(5_000).expect("Could not instantiate Sha512Params.");
+
+    // Hash the password
+    let hashed_password = sha512_crypt_b64(password.as_bytes(), salt.as_bytes(), &params)
+        .expect("Failed to compute sha512 password.");
+
+    let mut result = String::new();
+    // Pushing the prefix, then the salt, then the password.
+    result.push_str("$6$");
+    result.push_str(&salt);
+    result.push('$');
+    result.push_str(&hashed_password);
+    Ok(result)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn custom_salt_ok() {
```

Comment:
please add some negative tests as well.

Reply:
do you know how to mock things?

Reply:
This can actually never fail. I checked the library code, and the only failure scenarios are wrong rounds or if the bytes cannot be encoded as utf-8. Both can never happen in our case, so we should not return a result at all. Instead we should just unwrap, so it will panic if something goes bad in the libary.